x-pack/plugins/secops/public/containers/hosts/filter.tsx

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { connect } from 'react-redux';
+import { pure } from 'recompose';
+import { StaticIndexPattern } from 'ui/index_patterns';
+import {
+  hostsActions,
+  hostsSelectors,
+  KueryFilterQuery,
+  SerializedFilterQuery,
+  State,
+} from '../../store';
+import { getFilterComponent } from '../utils/filter_component';
+
+export interface HostsFilterArgs {
+  applyFilterQueryFromKueryExpression: (expression: string) => void;
+  filterQueryDraft: KueryFilterQuery;
+  isFilterQueryDraftValid: boolean;
+  setFilterQueryDraftFromKueryExpression: (expression: string) => void;
+}
+
+interface OwnProps {
+  children: (args: HostsFilterArgs) => React.ReactNode;
+  indexPattern: StaticIndexPattern;
+}
+
+interface HostsFilterReduxProps {
+  hostsFilterQueryDraft: KueryFilterQuery;
+  isHostFilterQueryDraftValid: boolean;
+}
+
+interface HostsFilterDispatchProps {
+  applyHostsFilterQuery: (filterQuery: SerializedFilterQuery) => void;
+  setHostsFilterQueryDraft: (filterQueryDraft: KueryFilterQuery) => void;
```

Comment:
The types for `applyHostsFilterQuery` and `setHostsFilterQueryDraft` should be `ActionCreator<{` as the other parts of the code are now using their `ActionCreator` types.

So it would be something like:

```ts
applyHostsFilterQuery: ActionCreator<KueryFilterQuery>;
setHostsFilterQueryDraft: ActionCreator<KueryFilterQuery>;
```